### PR TITLE
Fix broken links in MM docs

### PR DESCRIPTION
**Pull request by joaniefromtheblock**
# Description

Nightly check identified these links to be addressed ([Slack for ref](https://consensys.slack.com/archives/GKX130DMX/p1727764912083769)):

- https://community.optimism.io/docs/developers/build/json-rpc/ (Status: 404)
- https://community.optimism.io/docs/developers/bedrock/ (Status: 404)
- https://community.optimism.io/docs/developers/bedrock/#json-rpc (Status: 404)
- https://community.optimism.io/docs/developers/build/differences/ (Status: 404)
- https://developer.android.com/training/dependency-injection/hilt-android (Status: 0, redirect loop)
- https://developer.android.com/guide/components/processes-and-threads#IPC (Status: 0, redirect loop)
- https://developer.android.com/guide/components/aidl (Status: 0, redirect loop)
- https://framework.embarklabs.io/ (Status: 0, ENOTFOUND error)
- https://github.com/MetaMask/snaps-monorepo/tree/main/packages/examples/examples (Status: 404)
- https://github.com/ethereum/execution-apis/commits/assembled-spec/ (Status: 429, too many requests)

### CODE REVIEW

**Comment by joaniefromtheblock** [comment]
I think the link checker is handlinhg these links weirdly. I am not sure how to resolve: 
- https://developer.android.com/guide/components/processes-and-threads#IPC (Status: 0, redirect loop)
- https://developer.android.com/guide/components/aidl


I have tried updating but am still getting the same error.

**Comment by joaniefromtheblock** [comment]
Also, I cannot find the official docs page for Embark Labs, so am linked to the GitHub repo https://github.com/embarklabs/embark

File: services/reference/optimism/json-rpc-methods/index.md [hunk 1]
```diff
@@ -6,15 +6,15 @@ description: Supported standard Ethereum methods on Optimism network.
 # JSON-RPC methods
 
 The standard Ethereum methods documented here are supported by Infura on the Optimism network. For custom Optimism
-methods, please see [Optimism Ethereum JSON-RPC API documentation](https://community.optimism.io/docs/developers/build/json-rpc/) (Bedrock release).
+methods, see [Optimism Ethereum JSON-RPC API documentation](https://docs.optimism.io/builders/node-operators/json-rpc) (Bedrock release).
 
 :::info
 
-The Optimism Bedrock migration supports different JSON-RPC methods. Refer to the [Optimism Bedrock documentation](https://community.optimism.io/docs/developers/bedrock/#json-rpc) for more information.
+The Optimism Bedrock migration supports different JSON-RPC methods. Refer to the [Optimism Bedrock documentation](https://docs.optimism.io/stack/getting-started#the-op-stack-today) for more information.
```
Comment (bgravenorst) [inline services/reference/optimism/json-rpc-methods/index.md:13]:
```suggestion
The Optimism Bedrock migration supports different JSON-RPC methods. Refer to the [Optimism Bedrock documentation](https://docs.optimism.io/builders/node-operators/json-rpc) for more information.
```
Perhaps just link to the API docs here?

File: wallet/concepts/convenience-libraries.md [hunk 2]
```diff
@@ -14,7 +14,7 @@ The [MetaMask Ethereum provider API](wallet-api.md#ethereum-provider-api) is ver
 [Ethereum JSON-RPC](wallet-api.md#json-rpc-api) formatted messages, which is why
 some developers use a convenience library for interacting with the provider, such as
 [Ethers](https://www.npmjs.com/package/ethers), [web3.js](https://www.npmjs.com/package/web3), and
-[Embark](https://framework.embarklabs.io/).
+[Embark](https://github.com/embarklabs/embark).
```
Comment (bgravenorst) [inline wallet/concepts/convenience-libraries.md:17]: I'd remove the reference to Embark, even their X account hasn't been active in over 4 years.

**Review by bgravenorst — APPROVED** [review]
Minor feedback

File: services/reference/optimism/json-rpc-methods/index.md (hunk 1 above)
Comment (joaniefromtheblock) [reply services/reference/optimism/json-rpc-methods/index.md:13]: I am not sure the statement makes sense if we link to the same doc. What happened with the Bedrock migration? Are the Optimism Ethereum and Bedrock  JSON-RPC now the same?

Comment (bgravenorst) [reply services/reference/optimism/json-rpc-methods/index.md:13]: The Bedrock migration happened a while ago. I think even Optimism has removed references to Bedrock from it's pages. So perhaps just remove the adminition.

**Review by bgravenorst — APPROVED** [review]
lgtm

**Pull request merged**